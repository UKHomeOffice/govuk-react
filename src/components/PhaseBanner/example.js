import React from 'react';
import PhaseBanner from '.';

export default () => (
  <PhaseBanner level="alpha">
    This part of GOV.UK is being rebuilt &#8211;{' '}
    <a href="https://example.com">find out what that means</a>
  </PhaseBanner>
);
