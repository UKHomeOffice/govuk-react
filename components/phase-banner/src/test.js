import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import PhaseBanner from './';

const example = 'example';
const wrapper = <PhaseBanner level="beta">{example}</PhaseBanner>;

describe(PhaseBanner, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
