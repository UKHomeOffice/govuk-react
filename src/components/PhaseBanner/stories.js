import React from "react";
import { storiesOf } from "@storybook/react";

import PhaseBanner from ".";
import asAnchor from "../../hoc/asAnchor/index";

const AnchorLink = asAnchor("a");

storiesOf("PhaseBanner", module).add("Alpha phase tag", () => (
  <PhaseBanner level="alpha">
    This part of GOV.UK is being rebuilt &#8211;{" "}
    <AnchorLink href="https://example.com">find out what that means</AnchorLink>
  </PhaseBanner>
));

storiesOf("PhaseBanner", module).add("Beta phase tag", () => (
  <PhaseBanner level="beta">
    This part of GOV.UK is being rebuilt &#8211;{" "}
    <AnchorLink href="https://example.com">find out what that means</AnchorLink>
  </PhaseBanner>
));
