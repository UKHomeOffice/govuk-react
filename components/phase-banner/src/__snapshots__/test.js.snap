// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseBanner matches wrapper snapshot: wrapper mount 1`] = `
.glamor-1,
[data-glamor-1] {
  border-bottom: 1px solid #bfc1c3;
  box-sizing: border-box;
  padding-top: 10px;
  padding-bottom: 10px;
  font-family: "nta", Arial, sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 14px;
  line-height: 1.1428571429;
}

.glamor-1> a,
[data-glamor-1]> a {
  color: #005ea5;
}

.glamor-0,
[data-glamor-0] {
  display: inline-block;
  margin: 0 8px 0 0;
  padding: 2px 5px;
  font-family: "nta", Arial, sans-serif;
  font-weight: 700;
  font-size: 14px;
  line-height: 1.1428571429;
  text-transform: uppercase;
  letter-spacing: 1px;
  text-decoration: none;
  color: #ffffff;
  background-color: #005ea5;
  -webkit-font-mmoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

@media only screen and (min-width: 641px) {
  .glamor-1,
  [data-glamor-1] {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media only screen and (min-width: 641px) {
  .glamor-0,
  [data-glamor-0] {
    font-size: 16px;
    line-height: 1.25;
    padding-bottom: 0;
  }
}

<PhaseBanner
  level="beta"
>
  <glamorous(div)>
    <div
      className="glamor-1"
    >
      <PhaseBadge>
        <glamorous(strong)>
          <strong
            className="glamor-0"
          >
            beta
          </strong>
        </glamorous(strong)>
      </PhaseBadge>
      example
    </div>
  </glamorous(div)>
</PhaseBanner>
`;
